opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -115,6 +115,30 @@ void setupRecvInterface(const std::vector<std::tuple<int, int, char, int> >& lis
     }
 }
 #endif // HAVE_MPI
+
+/// Reclaim memory resources from CpGrid::InterfaceMap.  Used as custom
+/// deleter for std::shared_ptr<InterfaceMap>.
+struct FreeInterfaces
+{
+    void operator()([[maybe_unused]] Dune::CpGrid::InterfaceMap* interfaces) const
```

Comment:
I think #if around the definition rather than in the body would be clearer (and lets you skip maybe_unused when we know it won't be used which i personally prefer). It better communicates the do-nothing-without-mpi.

Reply:
> I think #if around the definition rather than in the body would be clearer

You mean as in
```C++
struct FreeIntefaces
{
#if MPI
    void operator()(...)
    {
        ...
    }
#endif
};
```
?  I don't think that'll work without some other changes.  In that case we'd need to either distinguish between the MPI and the non-MPI cases at [the call site](https://github.com/OPM/opm-grid/blob/3de8a0309ef3a622544f4d1fd1acc3bb257d409d/opm/grid/cpgrid/CpGrid.cpp#L150-L151) so as not to provide a custom deleter in the sequential case, or we'd need an alternative implementation of `FreeInterfaces::operator()` for the sequential case.

Reply:
i mean
```c++
#if HAVE_MPI
void operator(Dune::CpGrid::InterfaceMap* interfaces) const
{
..
}
#else
void operator()Dune::CpGrid::InterfaceMap*) const {}
#endif
```

Reply:
> i mean [an alternative implementation of `FreeInterfaces::operator()` for the sequential case.]

Right.  To be honest I'd prefer not to do that.

Reply:
fair enough.